src/playready.js

```diff
@@ -36,6 +36,16 @@ export const getMessageContents = (message) => {
     challenge = window.atob(challengeElement.childNodes[0].nodeValue);
   }
 
+  // If we failed to parse the xml the soap message might be encoded already.
+  // set the message data as the challenge and add generic SOAP headers.
+  if (xml.querySelector('parsererror')) {
+    headers = {
```

Comment:
Out of curiosity, is it possible to have preexisting headers from the previous logic that we would want to keep as well as adding these new headers?

Reply:
Since this logic will only be hit if there's an xml parsing error, there shouldn't be any valid headers to extract from the bad xml so I think replacing them here is safe.